src/Transformers/EnumTransformer.php

```diff
@@ -24,39 +26,41 @@ public function transform(ReflectionClass $class, string $name): ?TransformedTyp
             return null;
         }
 
+        $enum = (new ReflectionEnum($class->getName()));
+
+        if (! $enum->isBacked()) {
+            return null;
+        }
+
         return $this->config->shouldTransformToNativeEnums()
-            ? $this->toEnum($class, $name)
-            : $this->toType($class, $name);
+            ? $this->toEnum($enum, $name)
+            : $this->toType($enum, $name);
     }
 
-    protected function toEnum(ReflectionClass $class, string $name): TransformedType
+    protected function toEnum(ReflectionEnum $enum, string $name): TransformedType
```

Comment:
This only supports string enums

To support all types something like that should work:

```php
        $options = array_map(
            match ($enum->isBacked()) {
                true => match ((string)$enum->getBackingType()) {
                    'string' => fn ($enum) => sprintf('%s = "%s"', $enum->name, $enum->value),
                    default => fn ($enum) => sprintf('%s = %s', $enum->name, $enum->value),
                },
                false => fn ($enum) => $enum->name,
            },
            $enum->getConstants()
        );
```

Reply:
Hm, what's the point in handling non-backed enums? It makes no sense to convert them to TypeScript since they can not be used outside of PHP context anyways.

da8f449e5c1c466548fc70bd810e6fa3dbfd715e should add support for int backed enums.